scc/src/opts.rs

```diff
@@ -50,85 +50,189 @@ pub fn fix_args(args: Vec<String>) -> Vec<String> {
 #[derive(Clone, Debug)]
 pub struct Opts {
     pub script_paths: Vec<PathBuf>,
-    pub script_paths_file: Option<PathBuf>,
     pub cache_dir: Option<PathBuf>,
     pub optimize: bool,
     pub threads: u8,
-    pub no_warnings: bool,
+    pub warnings: Option<String>,
     pub no_testonly: bool,
     pub no_breakpoint: bool,
-    pub profile_off: bool,
+    pub profile: bool,
+    pub script_paths_file: Option<PathBuf>,
     pub cache_file: Option<PathBuf>,
 }
 
-fn toggle_options(name: &'static str, help: &'static str) -> impl Parser<bool> {
+impl Default for Opts {
+    fn default() -> Self {
+        Self {
+            script_paths: vec![],
+            cache_dir: None,
+            optimize: false,
+            threads: 1,
+            warnings: None,
+            no_testonly: false,
+            no_breakpoint: false,
+            profile: false,
+            script_paths_file: None,
+            cache_file: None,
+        }
+    }
+}
+
+fn no_space(name: &'static str, help: &'static str) -> impl Parser<Option<String>> {
     any::<String>(name)
         .help(help)
-        .guard(|s| s.starts_with('-'), "doesn't start with -")
+        .guard(
+            |s| s.starts_with(name.to_owned().as_str()),
+            "no_space doesn't start with -<name>",
+        )
         .parse(move |s| {
-            let Some(cur_name) = s.strip_prefix('-') else {
-                return Err(format!("{} is not a toggle option", s));
-            };
-            if cur_name != name {
-                Err(format!("{} is not a known toggle option name", cur_name))
+            // println!("no_space: {}", s);
+            s.strip_prefix(name.to_owned().as_str())
+                .ok_or(Err::<String, &str>("IDK1"))
+                .map_or(Err("IDK2"), |s| Ok(s.to_string()))
+        })
+        .anywhere()
+        .optional()
+        .catch()
+}
+
+fn toggle_options(name: &'static str, help: &'static str) -> impl Parser<Option<bool>> {
+    any::<String>(name)
+        .help(help)
+        .guard(|s| s.eq(name.to_owned().as_str()), "toggle_option isn't -<name>")
+        .parse(move |s| {
+            // println!("toggle_options: {}", s);
+            if s != name {
+                Err(format!("{} is not a known toggle option name", s))
             } else {
                 Ok(true)
             }
         })
+        .adjacent()
         .anywhere()
-        .fallback(false)
+        .optional()
+        .catch()
 }
 
-fn slong(tag_str: &'static str, arg_str: &'static str) -> impl Parser<String> {
-    let tag = any::<String>(tag_str).guard(move |s| s.as_str() == tag_str, "not arg name");
-    let value = positional::<String>(arg_str);
-    construct!(tag, value).anywhere().map(|pair| pair.1)
+fn equals_sign(name: &'static str, help: &'static str) -> impl Parser<Option<String>> {
+    any::<String>(name)
+        .help(help)
+        .guard(
+            |s| s.starts_with(format!("{}=", name.to_owned()).as_str()),
```

Comment:
```suggestion
            move |s| s.strip_prefix(name).and_then(|s| s.strip_prefix("=")).is_some(),
```
I'd avoiding formatting just to check a prefix